### PR TITLE
Response next previous

### PR DESCRIPTION
**Pull request by jaysonlarin**
With a Validic::Reponse, we can call the next or previous page simply by making a call to .next or .previous

- response.next
- response.previous

### CODE REVIEW

File: lib/validic/response.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
 module Validic
   class Response
+    include REST::Request
     attr_reader :summary, :records, :attributes
-    def initialize(summary, response)
-      @summary = summary
 
+    def initialize(summary, response, client = nil)
+      @summary = summary
+      @client = client if client
```
Comment (markphelps) [inline lib/validic/response.rb]: why the change to pass in the client?

File: lib/validic/response.rb [hunk 2]
```diff
@@ -11,5 +13,23 @@ def initialize(summary, response)
         @attributes = response
       end
     end
+
+    def next
+      extend Validic::REST::Utils
```
Comment (markphelps) [inline lib/validic/response.rb]: never seen `extend` on a method before. normally you put `extend` at the class level

File: lib/validic/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module Validic
-  VERSION = '0.4.1'
+  VERSION = '0.4.7'
```
Comment (markphelps) [inline lib/validic/version.rb]: this should probably be a minor version bump since we are adding new functionality instead of a bugfix bump. also why the bump from .1 to .7?

I would suggest `0.5.0` http://semver.org/

Comment (jaysonlarin) [inline lib/validic/version.rb]: will update this to 0.5.0

File: lib/validic/response.rb (hunk 1 above)
Comment (jaysonlarin) [inline lib/validic/response.rb]: some of the methods do not pass a client. we need a client for a faraday connection.

File: lib/validic/response.rb (hunk 2 above)
Comment (jaysonlarin) [inline lib/validic/response.rb]: i need the methods in Validic::REST::Utils in the next and previous methods but because of dependency issues. This is the only way i can come up at the moment.

File: lib/validic/response.rb (hunk 1 above)
Comment (markphelps) [inline lib/validic/response.rb]: it seems strange that the Reponse object would have a reference to the client that initiated the request, is there anyway to refactor this? also I do not see any changes reflecting faraday in this PR.

File: lib/validic/response.rb (hunk 2 above)
Comment (markphelps) [inline lib/validic/response.rb]: spoke with @jaysonlarin and tried to debug myself. we need to break the circular reference of `Response -> REST::Utils -> Response` in order for this to work correctly

File: lib/validic/version.rb [hunk 4]
```diff
@@ -1,3 +1,3 @@
 module Validic
-  VERSION = '0.4.1'
+  VERSION = '0.5.0'
```
Comment (markphelps) [inline lib/validic/version.rb]: we also need to add the `next` and `previous` methods to the README and explain the version bump

Comment (jaysonlarin) [inline lib/validic/version.rb]: done

File: lib/validic/response.rb (hunk 1 above)
Comment (jaysonlarin) [inline lib/validic/response.rb]: there is a connection method in response.rb

File: lib/validic/response.rb (hunk 2 above)
Comment (jaysonlarin) [inline lib/validic/response.rb]: this may require a revamp of what we have now, can we do the refactor on a later version?

File: lib/validic/response.rb (hunk 1 above)
Comment (markphelps) [inline lib/validic/response.rb]: yes, but you added that. my question is why? 

Above you say that faraday needs a client connection but Im not seeing where in the code that is true. maybe better stated: what does having a connection method on the Response object provide us the ability to do>

File: lib/validic/response.rb (hunk 2 above)
Comment (markphelps) [inline lib/validic/response.rb]: yeah I think thats best

File: lib/validic/response.rb (hunk 1 above)
Comment (jaysonlarin) [inline lib/validic/response.rb]: the connection gives us an ability to make the call for the next/previous set of records

**Comment by markphelps** [comment]
@jaysonlarin can you rebase these into a single commit before merging please? or if you would like me to I can

**Comment by jaysonlarin** [comment]
@markphelps thank you. Ill consolidate the commit.

**Comment by markphelps** [comment]
:+1: lgtm

**Comment by allcentury** [comment]
nice :+1:
